### ISSUE TITLE
CentOS Specific Install Instructions

### ISSUE THREAD

**Issue by mstackhouse** [jalan/pdftotext#33]
Following the instructions from this link:
[https://github.com/ropensci/pdftools#building-from-source][1]

**On CentOS**

On CentOS the `libpoppler-cpp` library is not included with the system so we need to build from source. Note that recent versions of poppler require C++11 which is not available on CentOS, so we build a slightly older version of libpoppler.

    # Build dependencies
    yum install wget xz libjpeg-devel openjpeg2-devel
    
    # Download and extract
    wget https://poppler.freedesktop.org/poppler-0.47.0.tar.xz
    tar -Jxvf poppler-0.47.0.tar.xz
    cd poppler-0.47.0
    
    # Build and install
    ./configure
    make
    sudo make install

By default libraries get installed in `/usr/local/lib` and `/usr/local/include`. On CentOS this is not a default search path so we need to set `PKG_CONFIG_PATH` and `LD_LIBRARY_PATH` to point R to the right directory:

    export LD_LIBRARY_PATH="/usr/local/lib"
    export PKG_CONFIG_PATH="/usr/local/lib/pkgconfig"

**Comment by jalan** [comment]
Does this apply only to CentOS 6? It looks like CentOS 7 has the required packages:
 - https://centos.pkgs.org/7/centos-x86_64/poppler-cpp-0.26.5-20.el7.x86_64.rpm.html
 - https://centos.pkgs.org/7/centos-x86_64/poppler-cpp-devel-0.26.5-20.el7.x86_64.rpm.html

**Comment by mstackhouse** [comment]
Yes - CentOS 7 has the required packages. I'll update the text and submit a
new PR if you'd like.

**Comment by dynamiccontent** [comment]
@mstackhouse @jalan how to install this on CentOS 8?

No match for argument: poppler-cpp-devel
No match for argument: python-devel
Error: Unable to find a match: poppler-cpp-devel python-devel

Using CentOS Linux release 8.1.1911 (Core)

Tried to install via instructions provided but it's messy and why I can't use the latest version of poppler? This is crazy...

**Comment by jalan** [comment]
CentOS moved some `devel` packages into the `PowerTools` repo, looks like. Just spun up CentOS 8 and this seems to work:

```
yum install dnf-plugins-core
yum config-manager --set-enabled PowerTools
yum install gcc-c++ pkgconfig poppler-cpp-devel python3-devel
pip3 install pdftotext
```

You can use the latest version of poppler if you want, but you won't find the latest version of just about anything in normal CentOS packages :man_shrugging:

**Comment by dynamiccontent** [comment]
@jalan Thank you very much for quick answer! I am up and running thanks to you. Would probably wasted many hours and still not arrive to the same result as you provided. You have been helpful! I am using your project with another project (SeedDMS).

**Comment by hoangph-1744** [comment]
@jalan thank you so much, solved my issue on centos-8 :bow:

**Comment by jalan** [comment]
Since this issue was originally about CentOS 6, which is EOL, I am closing it.
